pulsar-client-cpp/travis-build.sh

```diff
@@ -58,13 +58,18 @@ if [ "$3" = "all" -o "$3" = "dep" ]; then
   exec_cmd "pushd $1/ && wget https://github.com/google/protobuf/releases/download/v2.6.1/protobuf-2.6.1.tar.gz && popd";
   exec_cmd "pushd /usr/src/gtest && cmake . && make && cp *.a /usr/lib && popd";
   exec_cmd "pushd $1/ && tar xvfz $1/protobuf-2.6.1.tar.gz && pushd $1/protobuf-2.6.1 && ./configure && make && make install && popd && popd";
-fi           
+fi
 
 if [ "$3" = "all" -o "$3" = "compile" ]; then
   # Compile and run unit tests
   exec_cmd "pushd $2/pulsar-client-cpp && cmake . && make && popd";
   PULSAR_STANDALONE_CONF=$2/pulsar-client-cpp/tests/standalone.conf $2/bin/pulsar standalone &
-  pid=$!;
-  exec_cmd "sleep 10 && pushd $2/pulsar-client-cpp/tests && ./main && popd";
+  standalone_pid=$!;
+  PULSAR_STANDALONE_CONF=$2/pulsar-client-cpp/tests/authentication.conf $2/bin/pulsar standalone --zookeeper-port 2191 --bookkeeper-port 3191 --zookeeper-dir data2/standalone/zookeeper --bookkeeper-dir data2/standalone/zookeeper &
```

Comment:
The problem is the authentication plugin, not the broker offering encryption. If you have authentication enabled, the broker will require any client to provide the credentials, in any of the configured format. In that specific test, it's setting the standalone broker to accept TLS client authentication, so if you connect (even to the TLS port) without having a valid client certificate, the authentication will fail.